js/helpers.js

```diff
@@ -173,3 +179,12 @@ function getType(types, selectedType) {
   }
   throw new Error(`Unknown type: ${selectedType}`);
 }
+
+function calculateEMA(daysSet, idx, range, type) {
+  if (idx < 0) {
+    return daysSet[idx][type];
+  }
+
+  var k = 2/(range + 1);
+  return daysSet[idx][type] * k + daysSet[idx - 1][type] * (1 - k);
```

Comment:
I don't think this generates a valid EMA actually, EMA is an iterative function that operates on its output series.  For example `daysSet[idx - 1][type]` should reference the previous EMA output value not the source dataset.  That's why I generated the EMA dataset in a single pass

Reply:
I see. Makes sense. I don't mingle with data yet (I am considering doing it), so for now I'd prefer to go the "slow" route and see if the overhead is bareable.